sound/soc/sof/control.c

```diff
@@ -98,6 +98,31 @@ int snd_sof_volume_put(struct snd_kcontrol *kcontrol,
 	return 0;
 }
 
+int snd_sof_volume_info(struct snd_kcontrol *kcontrol,
+			struct snd_ctl_elem_info *uinfo)
+{
+	struct soc_mixer_control *mc =
+		(struct soc_mixer_control *)kcontrol->private_value;
+	struct snd_sof_control *scontrol = mc->dobj.private;
+	unsigned int channels = scontrol->num_channels;
+	int platform_max;
+
+	if (!mc->platform_max)
+		mc->platform_max = mc->max;
+
+	platform_max = mc->platform_max;
+
+	if (platform_max == 1 && !strstr(kcontrol->id.name, " Volume"))
```

Comment:
Cant do this string checking, best to have separate info() calls for volume and switch

Reply:
Yes, I will refine it